### ISSUE TITLE
Using LDAP,  Can't get Group info  in SW360 and Organization in liferay 

### ISSUE THREAD

**Issue by KoukiHama** [eclipse/sw360#710]
LDAP testing on liferay is all ok. I tried login by using ldap info. However, sw360 user don't belong to any kinds of organizations.  So, sw360 Preferences's group is empty and I can't see components.

 If I use https://github.com/sw360/ldapOrganizationMappingExt, Can I login as a user base on LDAP info?  Thanks.

LOG

`  
2019-11-06 07:22:04 ERROR SW360Assert:135 - User Department was empty in User(id:c9b71531e13b1079d41eb19xxxxxxxxx, revision:1-6c56xxxxac29efd582f10d398e1xxxxxxx, type:user, email:kouki1.hama@xxx.com, userGroup:USER, externalid: xxxxxxxx, fullname:kouki hama, givenname:kouki, lastname:hama, department:)
SW360Exception(why:null)
`

USE / sw360-5.1.0-M1 ff16072 /liferay 7

**Comment by mcjaeger** [comment]
No, sorry for late response, in mappings in the LDAP setting in Liferay you need to define a mapping to department.

The project you have cites does the following. Imagine your organisation has more detail in groups that what you would like to have in the sw360. Then, you cannot take the field 1:1, but you need to make a mapping. Example could be:

"Sales EMEA" -> "Sales"
"Sales Asia" -> "Sales"

if you have in your org the two groups for sales, EMEA and Asia, but you would like to see them in SW360 as one group.

@KoukiHama did you make any progress? so far?

**Comment by mcjaeger** [comment]
Just looked into it:

System settings -> LDAP -> Servers -> scroll down until there is the field for `User Custom Mappings `

the main screen looks like this:

<img width="757" alt="Screenshot 2019-11-28 at 22 06 25" src="https://user-images.githubusercontent.com/3418977/69831828-8572ac00-122b-11ea-9bbd-a96c7170e16e.png">

**Comment by KoukiHama** [comment]
Thank you for your advice. I solved this issues.

**Comment by dineshr93** [comment]
@KoukiHama I have deployed sw360 via docker chores.. Have you deployed via docker as well? Can you write a wiki on how to configure ldap?

**Comment by KoukiHama** [comment]
>Have you deployed via docker as well? 
Yes. However, when I used sw360 via dockers, I did not use ldap.
On the other hand, I plan to use sw360-docker with ldap setting.
If I can success to use ldap on sw360/docker, I would like to write these procedures.

**Comment by dineshr93** [comment]
@KoukiHama  Thanks ! I will look forward for it. :)

**Comment by dineshr93** [comment]
@KoukiHama I have installed sw360 via vagrant. Will you help me write a wiki for the ldap config https://github.com/sw360/ldapOrganizationMappingExt

**Comment by dineshr93** [comment]
Hi @KoukiHama, I get these following error after configuring ldap

```
2020-07-03 08:17:43.830 ERROR [http-nio-8080-exec-3][LDAPAuth:424] Problem accessing LDAP server
com.liferay.portal.kernel.exception.UserScreenNameException$MustValidate: Screen name ravi dinesh for user 36632 must validate with com.liferay.portal.kernel.security.auth.DefaultScreenNameValidator: The screen name cannot be an email address or a reserved word, such as postfix. It must contain only alphanumeric or the following special characters: -._.
        at com.liferay.portal.service.impl.UserLocalServiceImpl.validateScreenName(UserLocalServiceImpl.java:6863)
        at com.liferay.portal.service.impl.UserLocalServiceImpl.validate(UserLocalServiceImpl.java:6591)
        at com.liferay.portal.service.impl.UserLocalServiceImpl.addUserWithWorkflow(UserLocalServiceImpl.java:971)
        at com.liferay.portal.service.impl.UserLocalServiceImpl.addUser(UserLocalServiceImpl.java:785)
        at sun.reflect.GeneratedMethodAccessor869.invoke(Unknown Source)
```

have you witnessed the same by any change?

**Comment by dineshr93** [comment]
Hi @KoukiHama  Have you resolved this issue ?

I have successfully deployed the sw360 via source using https://github.com/eclipse/sw360/wiki/Deploy-Natively-11 long back.

I configured the ldap but still the department sync is not happening. 

I knew to resolve this there is https://github.com/sw360/ldapOrganizationMappingExt
custom user mapping **organization=department**

Have you tried this and worked? or you have some other way to resolve this?
